src/helpers/rateLimit.ts

```diff
@@ -42,3 +48,26 @@ export default rateLimit({
       })
     : undefined
 });
+
+export async function duplicateRequestLimit(req: Request, res: Response, next: NextFunction) {
+  if (!client) {
+    return next();
+  }
+
+  const value = hashedBody(req);
+  const [duplicate] = await client
+    .multi()
+    .SISMEMBER(DUPLICATOR_SET_KEY, value)
+    .SADD(DUPLICATOR_SET_KEY, value)
+    .exec();
```

Comment:
All to redis? will be costly? maybe fine

Reply:
Saving to redis should be fine, it's made for this. What may be costly is sha256 the whole body, don't know its cost yet

Reply:
Refactored to use `body.sig` instead, will improve performance